api_tests/lib2/actors/actor.ts

```diff
@@ -0,0 +1,271 @@
+            { name: alphaAsset.name, quantity: 0 },
+            betaAsset,
+        ]);
+
+        // this.expectedBalanceChanges.set(alphaAssetKind, -alphaAsset.quantity);
+        this.expectedBalanceChanges.set(betaAssetKind, betaAsset.quantity);
+        to.expectedBalanceChanges.set(alphaAssetKind, alphaAsset.quantity);
+        // to.expectedBalanceChanges.set(betaAssetKind, -betaAsset.quantity);
+
+        const comitClient: ComitClient = this.getComitClient();
+
+        const payload = {
+            alpha_ledger: alphaLedger,
+            beta_ledger: betaLedger,
+            alpha_asset: alphaAsset,
+            beta_asset: betaAsset,
+            alpha_expiry: defaultExpiryTimes().alpha_expiry,
+            beta_expiry: defaultExpiryTimes().beta_expiry,
+            peer: {
+                peer_id: await to.cnd.getPeerId(),
+                address_hint: await to.cnd
+                    .getPeerListenAddresses()
+                    .then(addresses => addresses[0]),
+            },
+            ...(await this.additionalIdentities(alphaAssetKind, betaAssetKind)),
+        };
+
+        this.swap = await comitClient.sendSwap(payload);
+        to.swap = new Swap(
+            to.wallets.bitcoin.inner,
+            to.wallets.ethereum.inner,
+            to.cnd,
+            this.swap.self
+        );
+        this.logger.debug("Created new swap at %s", this.swap.self);
+    }
+
+    public async accept() {
+        this.swap.accept(Actor.defaultActionConfig);
+    }
+
+    public async fund() {
+        if (!this.swap) {
+            throw new Error("Cannot fund inexistent swap");
+        }
+
+        this.logger.debug("Funding as part of swap @ %s", this.swap.self);
+        await this.swap.fund(Actor.defaultActionConfig);
+    }
+
+    public async redeem() {
+        if (!this.swap) {
+            throw new Error("Cannot redeem inexistent swap");
+        }
+
+        this.logger.debug("Redeeming as part of swap @ %s", this.swap.self);
+        await this.swap.redeem(Actor.defaultActionConfig);
+    }
+
+    public async assertSwapped() {
+        await new Promise(r => setTimeout(r, 1000));
+
+        this.logger.debug("Checking if swap @ %s is done", this.swap.self);
+
+        for (const [
+            assetKind,
+            expectedBalanceChange,
+        ] of this.expectedBalanceChanges.entries()) {
+            this.logger.debug(
+                "Checking that %s balance changed by %d",
+                assetKind,
+                expectedBalanceChange
+            );
+
+            const wallet = this.wallets[defaultLedgerForAsset(assetKind)];
+            const actualBalance = await wallet.getBalance();
+            const expectedBalance =
+                this.startingBalances.get(assetKind) + expectedBalanceChange;
+            const maximumFee = wallet.MaximumFee;
+
+            expect(actualBalance).to.be.at.least(expectedBalance - maximumFee);
+        }
+    }
+
+    private async additionalIdentities(
```

Comment:
This is definitely something that should be part of the sdk or API should be improved :(